crates/ruff/src/rules/pylint/rules/duplicate_values.rs

```diff
@@ -0,0 +1,44 @@
+use ruff_python_ast::comparable::ComparableExpr;
+use rustc_hash::FxHashSet;
+use rustpython_parser::ast::{self, Expr, Ranged};
+use std::hash::BuildHasherDefault;
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+
+use crate::checkers::ast::Checker;
+
+#[violation]
+pub struct DuplicateValues {
+    value: String,
+}
+
+impl Violation for DuplicateValues {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let DuplicateValues { value } = self;
+        format!("Duplicate value `{value}` in set.")
+    }
+}
+
+/// PLW0130
+/// "This message is emitted when a set contains the same value two or more times.",
+pub(crate) fn duplicate_values(checker: &mut Checker, elts: &Vec<Expr>) {
+    let mut seen_values: FxHashSet<ComparableExpr> =
+        FxHashSet::with_capacity_and_hasher(elts.len(), BuildHasherDefault::default());
```

Comment:
I think using `new` is the better choice here to avoid allocating in the case where none of the `elts` is a `constant` expression.

Reply:
@MichaReiser Thank you for the comment.
Do you mean switching to a normal `HashSet`, or using `default` instead of `with_capacity_and_hasher` ?

I can do
```rust
let mut seen_values: HashSet<ComparableExpr> = HashSet::new();
```
Or 
```rust
let mut seen_values: FxHashSet<ComparableExpr> = FxHashSet::default();
```
but I can't seem do use `new` with the `FxHashSet`.

Reply:
Hmm interesting, it must be because of some type constraint. `FxHashSet::default` should be fine.